### ISSUE TITLE
Documentation for `add_cryptol_defs` and `add_cryptol_eqs` could be better

### ISSUE THREAD

**Issue by msaaltink** [GaloisInc/saw-script#882]
Functions `add_cryptol_defs` and `add_cryptol_eqs` are somewhat mysterious.  I have a clear idea of what I think `add_cryptol_defs` should do, but cannot make it do anything.  For example:

    sawscript> let {{ f x = x + 0x0 }}
    sawscript> add_cryptol_eqs ["f"] basic_ss 
    saw: Stack trace:
    "add_cryptol_eqs" (<stdin>:1:1-1:16):
    Could not find definition: Cryptol.f

followed by saw helpfully exiting to the shell, giving me time to clear my mind before the next experiment.  Where might it be looking for definitions?

As for `add_cryptol_eqs`, which promises to "Add the named equality rules from the Cryptol module to the given
simplification rule set.", I do not know what these named equality rules might be.  Can I write my own?  Is there some built-in Cryptol module I should know about?

**Comment by brianhuffman** [comment]
Yes, there is a secret built-in `Cryptol.sawcore` module; you can find it in the sources here: https://github.com/GaloisInc/saw-core/blob/master/cryptol-saw-core/saw/Cryptol.sawcore

These primitives are of rather limited scope at the moment; `add_cryptol_eqs` is only for saw-core values defined in `Cryptol.sawcore` whose types *are* equations, i.e. `Eq` applied to something. I think there are only a couple of these that exist at the moment. `add_cryptol_defs` is similar, but you give it an ordinary-typed top-level name and it will give you an equation or equations derived from that constant's saw-core definition. (`add_prelude_eqs` and `add_prelude_defs` are the equivalents for `Prelude.sawcore`.) We should probably try to generalize/improve the user interface for these, so that we don't need two variants of each, and also so that users have a way to browse the list of definitions/rules that are available.

This thing that you're trying to do, automatically deriving a rewrite rule from a Cryptol definition, is something that we don't support at the moment. That's a feature that we might want to add. (Currently we can only use `unfolding`/`unfold_term` with definitions derived from Cryptol, but not the rewriter unless you state and prove the rules explicitly.)
